src/core/annotation.js

```diff
@@ -752,6 +754,42 @@ var TextWidgetAnnotation = (function TextWidgetAnnotationClosure() {
   return TextWidgetAnnotation;
 })();
 
+var ButtonWidgetAnnotation = (function ButtonWidgetAnnotationClosure() {
+  function ButtonWidgetAnnotation(params) {
+    WidgetAnnotation.call(this, params);
+
+    this.data.pushbutton = this.hasFieldFlag(AnnotationFieldFlag.PUSHBUTTON);
+    this.data.radio = !this.data.pushbutton &&
+                      this.hasFieldFlag(AnnotationFieldFlag.RADIO);
+    var fieldValue = Util.getInheritableProperty(params.dict, 'V');
+    if (isName(fieldValue)) {
```

Comment:
What happens if it's not a `Name`? It happens very often that PDF files do not adhere to the specification entirely, so if they do not provide a `Name` here, then `this.data.fieldValue` will be used as-is after the other PR is merged. Should we make it empty in such a case?